File: MPI/solutions/mpi_pnt2pnt.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char* argv[])
{
	MPI_Init(&argc,&argv);

	// Get local rank
	int myRank;
	MPI_Comm_rank(MPI_COMM_WORLD, &myRank);

	if(myRank == 0) // Work for process with rank 0
	{
		int val_send = 10;
		MPI_Send(&val_send, 1, MPI_INT, 1, 17, MPI_COMM_WORLD);
	}
	else if(myRank == 1) // Work for process with rank 1
	{
		int val_recv;
		MPI_Status status;
		MPI_Recv(&val_recv, 1, MPI_INT, 0, 17, MPI_COMM_WORLD, &status);
		printf("Rank %d received value %d\n", myRank, val_recv);
	}
	else // Other ranks do nothing
	{
	}

	MPI_Finalize();
}

File: MPI/exercises/mpi_pnt2pnt.f
      program pnt2pnt

        use mpi
        implicit none

        integer :: ierr, myRank, status(MPI_STATUS_SIZE)
        integer :: val_send, val_recv

        call MPI_INIT(ierr)

c       Get local rank
        call MPI_COMM_RANK(MPI_COMM_WORLD, myRank, ierr)

        if (myRank == 0) then
          val_send = 10
          call MPI_SEND(val_send, 1, MPI_INTEGER, XX, AA,
     &                  MPI_COMM_WORLD, ierr)
        else if (myRank == 1) then
          call MPI_RECV(val_recv, 1, MPI_INTEGER, YY, BB, 
     &                  MPI_COMM_WORLD, status, ierr)
          print *,"Rank ",myRank," received value ",val_recv
        endif
        
        call MPI_Finalize(ierr)

      end program
